core/src/impl/Kokkos_Error.cpp

```diff
@@ -102,11 +102,13 @@ void Experimental::RawMemoryAllocationFailure::print_error_message(
   o << "  (The allocation mechanism was ";
   switch (m_mechanism) {
     case AllocationMechanism::StdMalloc: o << "standard malloc()."; break;
```

Comment:
How does this work if KOKKOS_ENABLE_DEPRECATED_CODE_4 is off? Isn't the entire AllocationMechanism thing removed if DEPRECATED_CODE_4 is not enabled?

Reply:
You are looking at the wrong place, the relevant is `core/src/impl/Kokkos_Error.hpp`, not `core/src/Kokkos_HostSpace.hpp`. Both files contain `AllocationMechanism` enums but they are different. Note that we test with `KOKKOS_DEPRECATED_CODE_4=OFF` in our CI, see https://github.com/kokkos/kokkos/blob/1a3ea28f6e97b4c9dd2c8ceed53ad58ed5f94dfe/.jenkins#L56-L92 for example.

Reply:
Ok that seems somewhat unfortunate naming but ok.